qiskit_nature/problems/sampling/protein_folding/bead_distances/distance_map.py

```diff
@@ -0,0 +1,150 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class that stores distances between beads of a peptide as qubit operators."""
+from typing import Union, Tuple, DefaultDict, Dict
+import numpy as np
+
+from qiskit.opflow import PauliSumOp, PauliOp, OperatorBase
+
+from .distance_map_builder import DistanceMapBuilder
+from ..peptide.beads.base_bead import BaseBead
+from ..peptide.pauli_ops_builder import _build_full_identity
+from ..peptide.peptide import Peptide
+from ..qubit_utils.qubit_fixing import _fix_qubits
+
+
+class DistanceMap:
+    """Stores distances between beads of a peptide as qubit operators."""
+
+    def __init__(self, peptide: Peptide):
+        """
+        Args:
+            peptide: A Peptide object that includes all information about a protein.
+        """
+        self._peptide = peptide
+        self._distance_map, self._num_distances = DistanceMapBuilder()._create_distance_qubits(
+            peptide
+        )
+
+    def __getitem__(self, position: Tuple[BaseBead, BaseBead]) -> OperatorBase:
+        item1, item2 = position
+        return self._distance_map[item1][item2]
+
+    @property
+    def peptide(self) -> Peptide:
+        """Returns a peptide."""
+        return self._peptide
+
+    @property
+    def distance_map(self) -> DefaultDict[BaseBead, Dict[BaseBead, OperatorBase]]:
+        """Returns a distance map."""
+        return self._distance_map
+
+    @property
+    def num_distances(self) -> int:
+        """Returns the number of distances calculated."""
+        return self._num_distances
+
+    def _first_neighbor(
+        self,
+        peptide: Peptide,
+        lower_bead_ind: int,
+        is_side_chain_upper: int,
+        upper_bead_ind: int,
+        is_side_chain_lower: int,
```

Comment:
I am not sure where I addressed this before but I find the ordering weird here: `lower, upper, upper, lower`. Please unify this.
Also: the docstring should list the arguments in the same order in which they appear.

Ensure to update the order in places where this method is called.